PR for BXMSDOC-7240: Single-source and add "Customizing project preferences" chapter in the upstream documents. (#108)
kaldesai/kie-docs: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/General/customizing-project-preferences-proc.adoc

[id='customizing-project-preferences-proc_{CONTEXT}']
= Customizing project preferences

> A project stores assets and is part of a space. A space can hold multiple projects.

Reviewer: ```suggestion
In the {CENTRAL}, a project is a part of your space and stores the related assets. You can add multiple projects in a space.
```

Author: Done

> For example, an organization has many departments, such as HR, Payroll, Engineering, R&D, and so on. Each department maps to a space and every department can have its own projects.

Reviewer: ```suggestion
For example, an organization includes various departments, such as HR, Payroll, Engineering, and R&D. Each department can be mapped to a space in {CENTRAL}, along with adding respective projects. 
```

Author: Done

> You can create a new project from scratch or clone projects from an existing Git repository.

Reviewer: ```suggestion
You can create a new project or clone projects from an existing Git repository.
```

Author: Done.

> .Procedure

Reviewer 2: IMHO, this is a bit confusing. The above paragraph is about the possibility to create/import project and then there's "Procedure". To me, it sounds like this should be a procedure for creating/importing projects. Maybe, I would add some sentence add the end of the above paragraph like "As a user, you can customize the project settings in the business central." and then Procedure:... to show the user that the procedure is about the project settings customization

Author: Done. I have included a statement about customizing the projects in the introduction part.

> . In {CENTRAL}, select the *Admin* icon in the upper-right corner of the screen and select *Projects*.

Reviewer: ```suggestion
. In {CENTRAL}, select the *Admin* icon in the upper-right corner and select *Projects*.
```

Author: Done

+

> The *Projects* page opens.

Reviewer 2: This can be confusing as in the Business central menu there's "Projects". I am not sure about the "Projects page" words together. Would it be possible to rephrase this? Maybe use "settings" instead of page?

Author: This is fluff. I have removed it from the content. I think it is not necessary to mention that it opens. The user directly gets access to the **Project Preferences** panel.

> . In the *Project Preferences* panel, click the preference you want to modify. The following preferences are available:

Reviewer: ```suggestion
. In the *Project Preferences* panel, select the preference you want to modify. The project preferences include:
```

Author: Done

+

> * *Project Importing*: This preference has the following property:

Reviewer: Which property are we mentioning here? 
```suggestion
* *Project Importing*: This preference consists of the following property:
```

Author: Done

> ** Select the *Allow multiple projects to be imported on cluster* checkbox to import multiple projects at the same time on a cluster.

Reviewer: ```suggestion
** Select the *Allow multiple projects to be imported on cluster* to import multiple projects on a cluster.
```

Author: Done

> * *File exporting*: This preference has the following properties:

Reviewer: ```suggestion
* *File exporting*: This preference consists of the following properties:
```

Author: Done

+
[caption="Table 1. "]
.File exporting properties
[%header,cols=2]
[%autowidth]
|===
|Field |Description

|PDF orientation |Determines whether the PDF orientation is portrait or landscape.

> |PDF units |Determines whether the PDF unit is _PT_, _MM_, _CN_ or _IN_.

Reviewer: We can also mention what the unit _PT_ or _MM_ stands for. 
```suggestion
|PDF units |Determines whether the PDF unit is _PT_, _MM_, _CN_, or _IN_.
```

|PDF page format |Determines whether the PDF page format is _A[0-10]_, _B[0-10]_, or _C[0-10]_.
|===
+

> * *Spaces*: This preference has the following properties:

Reviewer: ```suggestion
* *Spaces*: This preference consists of the following properties:
```

+
[caption="Table 2. "]
.Spaces properties
[%header,cols=2]
[%autowidth]
|===
|Field |Description

|Name |The default name of the space that is created automatically if none exists.

|Owner |The default owner of the space that is created automatically if none exists.

|Group ID |The default group ID of the space that is created automatically if none exists.

|Alias (in singular) |Determines the customized alias (singular) of the space.

|Alias (in plural) |Determines the customized alias (plural) of the space.
|===
+

> * *Default values*: This preference has the following properties:

Reviewer: ```suggestion
* *Default values*: This preference consists of the following properties:
```

+
[caption="Table 3. "]
.Default values properties
[%header,cols=2]
[%autowidth]
|===
|Field |Description

> |Version |The default version number of a project when creating projects using the Quick setup option.

Reviewer: Is Quick available in UI?

Author: Actually not. I will remove it.

|Description |The default description of a project when creating projects using the Quick setup option.

|Branch |The default branch to be used when using a Git repository.

|Assets Per Page| Used to customize the number of assets per page in the project. The default value is `15`.

|===
+

> * *Advanced GAV preferences*: This preference has the following properties:

Reviewer: ```suggestion
* *Advanced GAV preferences*: This preference consists of the following properties:
```

+
[caption="Table 4. "]
.Advanced GAV preference properties
[%header,cols=2]
[%autowidth]
|===
|Field |Description

> |Disable GAV conflict check? |Determines whether to enable or disable the GAV conflict check. Disabling this feature allows projects to have the same GAV (Group ID, Artifact, Version).

Reviewer: ```suggestion
|Disable GAV conflict check? |Determines whether to enable or disable the GAV conflict check. Disabling this checkbox enables the projects to contain the same GAV (group ID, artifact, and version).
```

Author: Done

> |Allow child GAV edition? |Determines whether to allow child or sub-projects to have GAV edition.

Reviewer: ```suggestion
|Allow child GAV edition? |Determines whether to allow child or subprojects to contain GAV edition.
```

Author: Done

|===
+

> NOTE: Duplicate GAV detection is disabled for projects in *Development Mode*. To enable duplicate GAV detection for a project in {CENTRAL}, go to project *Settings* -> *General Settings* -> *Version* and toggle the *Development Mode* option to *OFF* (if applicable).

Reviewer: ```suggestion
NOTE: Duplicate GAV detection is disabled for projects in the development mode. To enable duplicate GAV detection for a project in {CENTRAL}, go to project *Settings* -> *General Settings* -> *Version* and toggle the *Development Mode* option to *OFF* (if applicable).
```

Author: Done.
